### DISCUSSION TITLE
[Question] Contributing to Parachute

### DISCUSSION THREAD

**Discussion by da-viper** [tcorreabr/Parachute#65] [Q&A] [answered]
Hi there, what ide or text editors do you use for writing this kwin script, how do you test the changes you have made to the project, 

is there any other thing that i need in other to contribute ? 

thanks

**Comment by tcorreabr** [comment, marked as answer]
Ideally, you should use an ide with integrated QML debbuger, like Qt Creator. And I even managed to use it in the early stages of the project. When I started using KWin scripts dependencies, I thought it was better to migrate to a code editor, like VS Code. I still use Qt Creator to develop new code snippets, but most of the testing is done by trial and error in VS Code.

**Reply by da-viper** [reply]
sorry for the late reply 

in other to have a project in Qt-creator you need either a cmakelist file or a qmake for it to work  

do you have an existing CMakelist/ qmake file ? 
 
also what extension do you use in vscode because there is no extension that supports qml ? 

thanks

**Reply by tcorreabr** [reply]
> in other to have a project in Qt-creator you need either a cmakelist file or a qmake for it to work
> 
> do you have an existing CMakelist/ qmake file ?

You can use Qt Creator as an editor for qml files. But since we can't run/debug a KWin script through it, I see no reason to create a project. You should run the command "kwin_x11 --replace &" to see your changes working.

> also what extension do you use in vscode because there is no extension that supports qml ?

I am using https://marketplace.visualstudio.com/items?itemName=seanwu.vscode-qt-for-python for some qml support on vscode.
